### PR TITLE
Add note about NEXT_DATA hydration

### PR DESCRIPTION
**Pull request by MaedahBatool**
Missing documentation around `NEXT_DATA` hydration when using `getServerSideProps()`. This PR warn users that the result of `getServerSideProps()/getStaticProps` is also available on the client-side due to React Hydration so they should not use any sensitive information in the props.

## Bug

- [x] Related issues linked using [fixes number](https://github.com/vercel/documentation/issues/418)
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm lint`
- [ ] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

File: docs/api-reference/data-fetching/get-initial-props.md [hunk 1]
```diff
@@ -67,6 +67,7 @@ For the initial page load, `getInitialProps` will run on the server only. `getIn
 
 - `getInitialProps` can **not** be used in children components, only in the default export of every page
 - If you are using server-side only modules inside `getInitialProps`, make sure to [import them properly](https://arunoda.me/blog/ssr-and-server-only-modules), otherwise it'll slow down your app
+- Note that irrespective of rendering type, any `props` will be passed to the page component and can be viewed on the client-side. Make sure that you don't pass any sensitive information via `props`.
```
Comment (ijjk) [inline docs/api-reference/data-fetching/get-initial-props.md:70]:
```suggestion

> Note that irrespective of rendering type, any `props` will be passed to the page component and can be viewed on the client-side. Make sure that you don't pass any sensitive information via `props`.
```

File: docs/basic-features/data-fetching/get-static-props.md [hunk 2]
```diff
@@ -14,6 +14,8 @@ export async function getStaticProps(context) {
 }
 ```
 
+> Note that irrespective of rendering type, any `props` will be passed to the page component and can be viewed on the client-side. Make sure that you don't pass any sensitive information via `props`.
```
Comment (ijjk) [inline docs/basic-features/data-fetching/get-static-props.md:17]:
```suggestion
> Note that irrespective of rendering type, any `props` will be passed to the page component and can be viewed on the client-side in the initial HTML. This is to allow the page to be [hydrated](https://reactjs.org/docs/react-dom.html#hydrate) correctly. Make sure that you don't pass any sensitive information that shouldn't be available on the client in `props`.
```

File: docs/basic-features/data-fetching/get-server-side-props.md [hunk 3]
```diff
@@ -14,6 +14,8 @@ export async function getServerSideProps(context) {
 }
 ```
 
+> Note that irrespective of rendering type, any `props` will be passed to the page component and can be viewed on the client-side. Make sure that you don't pass any sensitive information via `props`.
```
Comment (ijjk) [inline docs/basic-features/data-fetching/get-server-side-props.md:17]:
```suggestion
> Note that irrespective of rendering type, any `props` will be passed to the page component and can be viewed on the client-side in the initial HTML. This is to allow the page to be [hydrated](https://reactjs.org/docs/react-dom.html#hydrate) correctly. Make sure that you don't pass any sensitive information that shouldn't be available on the client in `props`.
```

File: docs/api-reference/data-fetching/get-initial-props.md [hunk 4]
```diff
@@ -68,6 +68,8 @@ For the initial page load, `getInitialProps` will run on the server only. `getIn
 - `getInitialProps` can **not** be used in children components, only in the default export of every page
 - If you are using server-side only modules inside `getInitialProps`, make sure to [import them properly](https://arunoda.me/blog/ssr-and-server-only-modules), otherwise it'll slow down your app
 
+> Note that irrespective of rendering type, any `props` will be passed to the page component and can be viewed on the client-side. Make sure that you don't pass any sensitive information via `props`.
```
Comment (ijjk) [inline docs/api-reference/data-fetching/get-initial-props.md:71]:
```suggestion
> Note that irrespective of rendering type, any `props` will be passed to the page component and can be viewed on the client-side in the initial HTML. This is to allow the page to be [hydrated](https://reactjs.org/docs/react-dom.html#hydrate) correctly. Make sure that you don't pass any sensitive information that shouldn't be available on the client in `props`.
```

**Review by ijjk — APPROVED** [review]
I think we can expand on this and potentially have a separate section specifically about hydration although this seems like a good initial note to have, thanks!

**Comment by MaedahBatool** [comment]
> I think we can expand on this and potentially have a separate section specifically about hydration although this seems like a good initial note to have, thanks!

Yes, that's a nice idea. We can def. work something around this.

File: docs/api-reference/data-fetching/get-initial-props.md [hunk 5]
```diff
@@ -68,6 +68,8 @@ For the initial page load, `getInitialProps` will run on the server only. `getIn
 - `getInitialProps` can **not** be used in children components, only in the default export of every page
 - If you are using server-side only modules inside `getInitialProps`, make sure to [import them properly](https://arunoda.me/blog/ssr-and-server-only-modules), otherwise it'll slow down your app
 
+> Note that irrespective of rendering type, any `props` will be passed to the page component and can be viewed on the client-side in the initial HTML. This is to allow the page to be [hydrated](https://reactjs.org/docs/react-dom.html#hydrate) correctly. Make sure that you don't pass any sensitive information that shouldn't be available on the client in `props`.
```
Comment (leerob) [inline docs/api-reference/data-fetching/get-initial-props.md:71]: Let's keep the format consistent with other notes. **Note:** 

I think we can drop "that irrespective of rendering type"

*Note:** Any `props` returned in `getInitialProps` will be passed to the...

"Make sure that you don't pass any" to just "Don't pass any..."

**Pull request merged**